client/coral-admin/src/actions/settings.js

```diff
@@ -21,10 +23,16 @@ export const fetchSettings = () => dispatch => {
     });
 };
 
+// for updating top-level settings
 export const updateSettings = settings => {
   return {type: SETTINGS_UPDATED, settings};
 };
 
+// this is a nested property, so it needs a special action.
+export const updateWordlist = (listName, list) => {
```

Comment:
The benefit here is grouping of concerns. This is the parallel of normalizing the database if we were using a SQL style database. By grouping like data together, we reduce duplication of namespaces and I think, aids in focusing the data models to their use cases and simplifying the API.

Imagine if we had a endpoint specifically for updating only the wordlist data, we could simple expose an `PUT /api/settings/wordlist` endpoint rather than having to rely on a `PATCH /api/settings`.

I understand that some additional code may be required on the front end in order to reduce the data to a form that is compatible with the front end data stores, but I don't think that it is unwarranted considering the vast implications that it leads to when we consider other nested objects, like `metadata`.

If we later decided to change our datastore to something like postgres, then would it not make sense to contain wordlist data in a table called `settings_wordlist`? Then the outgoing data would have to take the form of an embedded document to ensure Retrieve/Update consistency.

To me, it is essential to establish name-spacing with subdocuments.

RE: @jde, @davidgljay, @riley